clients/src/main/java/org/apache/kafka/clients/ClientRequest.java

```diff
@@ -126,7 +115,8 @@ public long sendTimeMs() {
         return sendTimeMs;
     }
 
-    public void setSendTimeMs(long sendTimeMs) {
+    public void setSend(Send send, long sendTimeMs) {
```

Comment:
This feels a bit awkward. Instead of passing in send separately, could we pass in body in the constructor and generate Send in the constructor from body but not storing body in ClientRequest?